packages/common/src/store/pages/chat/sagas.ts

```diff
@@ -133,29 +133,51 @@ function* doMarkChatAsRead(action: ReturnType<typeof markChatAsRead>) {
     const chat = yield* select((state) => getChat(state, chatId))
     if (!chat || chat?.unread_message_count > 0) {
       yield* call([sdk.chats, sdk.chats.read], { chatId })
-      if (chat) {
-        yield* put(upsertChat({ chat: { ...chat, unread_message_count: 0 } }))
-      }
+      yield* put(markChatAsReadSucceeded({ chatId }))
     }
   } catch (e) {
     console.error('markChatAsReadFailed', e)
+    yield* put(markChatAsReadFailed({ chatId }))
   }
 }
 
+let tempMessageIdCounter = 1
```

Comment:
Will probably iterate on this - might have SDK expose the cuid() generation and use that... but this works for now...